vllm/worker/hpu_model_runner.py

```diff
@@ -979,7 +1005,7 @@ def _prepare_decode(
                                        dtype=torch.long,
                                        device='cpu')
 
-        num_decode_tokens = sum(seq_lens)
+        num_decode_tokens = len(seq_lens)
```

Comment:
Why make this change?

Reply:
`seq_lens` represents the total length of each sequence, whereas the `num_decode_tokens` should be derived from the query length of each sequence, aligning with the logic used in the GPU model runner.

Reply:
Make sense to me, but I think this seems to be a general bug fixing in that case? @kzawora-intel  @michalkuligowski , may you take a look at this?